### ISSUE TITLE
[progress][meter] accessibility: NVDA and JAWS don't announce the programmatic label

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4184]
# Bug report

## Current behavior

Unfortunately, screen reader support for the progress and meter roles is still quite inconsistent. 

During testing, both NVDA and JAWS announced "Progress bar #%"  and did not announce the visible label that is referenced using `aria-labelledby`. This makes the progress bars lose all meaning because they are not labelled.

## Expected behavior

Progress bar programmatic (and visible) labels should be announced by screen readers. 

Although I suspect this may be a problem with the screen readers themselves, the current experience will be very poor for people using a screen reader when they encounter this BaseUI component, particularly if, for example, several meter components are used on a page, each with distinct labels.

## Reproducible example

Steps to reproduce: 
1. Navigate to the [Meter component demo](https://base-ui.com/react/components/meter)
2. Turn on either NVDA or JAWS on Windows
3. Use the Arrow keys to move through the component demo (reading mode)
4. Note the progress bar role and value is announced without announcing the visible label of "Storage used"
5. Test again with VoiceOver on mac and note the difference. The visible label is announced as expected along with the role and value.

## Base UI version

v1.2.0

## Which browser are you using?

Chrome

## Which OS are you using?

Windows

## Which assistive tech are you using (if applicable)?

NVDA and JAWS

## Additional context

Making the element with `role="meter"` focusable using `tabindex="0"` would fix the announcement and improve this, even though this is not meant to be a focusable element. In this case, this would be a valid improvement, as the minor trade-off of an extra tab stop would fix a serious issue for two widely used screen readers.

**Comment by mj12albert** [comment]
I tested this in NVDA, it won't even announce an `aria-label` 🥲

**Comment by mj12albert** [comment]
For JAWS, there's a bug on their side: https://github.com/mui/base-ui/pull/4200#issuecomment-3964718820

For NVDA, they have a few issues too but we can fix it on our side, preview: https://master--base-ui.netlify.app/react/components/progress

Tested it with NVDA+Edge:

https://github.com/user-attachments/assets/9390fbcf-f381-44e8-ae8f-b001b1e3d41c
